src/Compilers/CSharp/Portable/BoundTree/Expression.cs

```diff
@@ -202,22 +202,23 @@ private static IArgument DeriveArgument(int parameterIndex, int argumentIndex, I
 
         private static IOperation CreateParamArray(IParameterSymbol parameter, ImmutableArray<BoundExpression> boundArguments, int firstArgumentElementIndex, SyntaxNode invocationSyntax)
         {
-            if (parameter.Type.TypeKind == TypeKind.Array)
+            ArrayBuilder<IOperation> builder = ArrayBuilder<IOperation>.GetInstance(boundArguments.Length - firstArgumentElementIndex);
+            for (int index = firstArgumentElementIndex; index < boundArguments.Length; index++)
             {
-                IArrayTypeSymbol arrayType = (IArrayTypeSymbol)parameter.Type;
-                ArrayBuilder<IOperation> builder = ArrayBuilder<IOperation>.GetInstance(boundArguments.Length - firstArgumentElementIndex);
+                builder.Add(boundArguments[index]);
+            }
 
-                for (int index = firstArgumentElementIndex; index < boundArguments.Length; index++)
-                {
-                    builder.Add(boundArguments[index]);
-                }
-                var paramArrayArguments = builder.ToImmutableAndFree();
+            var paramArrayArguments = builder.ToImmutableAndFree();
 
+            if (parameter.Type.TypeKind == TypeKind.Array)
+            {
+                IArrayTypeSymbol arrayType = (IArrayTypeSymbol)parameter.Type;
+                
                 // Use the invocation syntax node if there is no actual syntax available for the argument (because the paramarray is empty.)
                 return new ArrayCreation(arrayType, paramArrayArguments, paramArrayArguments.Length > 0 ? paramArrayArguments[0].Syntax : invocationSyntax);
             }
 
-            return new InvalidExpression(invocationSyntax);
+            return new InvalidExpression(invocationSyntax, paramArrayArguments);
```

Comment:
So then are we fine with having no IOperation nodes in the IOperation tree? I though InvalidExpression and BoundBadExpression were just created to wrap bound nodes in presence of semantic errors so that the bound tree still has all the bound nodes. I will discuss this offline to understand better.